### ISSUE TITLE
Support for Python 3.12

### ISSUE THREAD

**Issue by dantetemplar** [mittagessen/kraken#613]
When I was trying to add kraken package to my project on Python 3.12 I got error:

```
poetry add kraken
Using version ^5.2.5 for kraken

Updating dependencies
Resolving dependencies... (0.1s)

The current project's supported Python range (>=3.10,<3.13) is not compatible with some of the required packages Python requirement:
  - kraken requires Python <=3.11.99,>=3.8, so it will not be satisfied for Python >3.11.99,<3.13

Because no versions of kraken match >5.2.5,<6.0.0
 and kraken (5.2.5) requires Python <=3.11.99,>=3.8, kraken is forbidden.
So, because innohassle-search-experiments depends on kraken (^5.2.5), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties
    
    For kraken, a possible solution would be to set the `python` property to ">=3.10,<=3.11.99"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```

Maybe package dependency on Python should be extended.

**Comment by dantetemplar** [comment]
If I can help somehow - tag me and give kind off instructions.

**Comment by mittagessen** [comment]
Yes, there are a couple of dependencies that don't have 3.12 and some
people are rquired to run kraken on python 3.8 which some
3.12-compatible dependency release have already deprecated. There won't
be a bump up to 3.12 until 3.8 is EOL the end of this year at least.

**Comment by dantetemplar** [comment]
> Yes, there are a couple of dependencies that don't have 3.12 and some people are rquired to run kraken on python 3.8 which some 3.12-compatible dependency release have already deprecated. There won't be a bump up to 3.12 until 3.8 is EOL the end of this year at least.

Okay, let's freeze this issue until [2024-10](https://devguide.python.org/versions/).

**Comment by amitdo** [comment]
https://discuss.python.org/t/python-3-8-is-now-officially-eol/66983

**Comment by mittagessen** [comment]
On 24/10/14 08:56AM, Amit D. wrote:
> https://discuss.python.org/t/python-3-8-is-now-officially-eol/66983

I'll remove 3.8, add 3.12, update any dependencies and make sure
everything works, and tag a new releasein the next couple of days.

**Comment by mittagessen** [comment]
I'm in the process of deprecating 3.8 and bumping up most dependencies (see PR #653). Unfortunately, coremltools still doesn't have 3.12 support so I won't be able to enable that one for now.

**Comment by amitdo** [comment]
https://github.com/apple/coremltools/issues/2129

**Comment by amitdo** [comment]
https://pypi.org/project/coremltools/#data

coremltools 8.1 supports python 3.12.

**Comment by mittagessen** [comment]
OK, that let's us upgrade once I'm finished with the shapely 2.x stuff.

**Comment by mittagessen** [comment]
I've just merged the PR enabling 3.12 support and tagged a new 5.3.0 release.
